.github/workflows/ci.yaml

```diff
@@ -0,0 +1,54 @@
+name: Build snap and run tests
+
+on:
+  workflow_call
+
+jobs:
+  check:
+    name: Code checks
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version:
+          - '3.9'
+          - '3.10'
+    steps:
+      - id: checkout
+        name: Checkout repo
+        uses: actions/checkout@v3
+      - id: setup-python
+        name: Setup Python
+        uses: actions/setup-python@v2.2.2
+        with:
+          python-version: ${{matrix.python-version}}
+          architecture: x64
+      - id: code-checks
+        name: Make sure the code follows conventions and tests correctly
+        run: |
+          make checks
+
+  test:
+    name: Unit and Integration Tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    needs:
+      - check
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - id: setup-python
+        name: Setup Python
+        uses: actions/setup-python@v2.2.2
+        with:
+          python-version: ${{matrix.python-version}}
```

Comment:
I'm not sure we are in a matrix here. Unless integration tests are rather time consuming (more than 10 min), it could still make sense to run these against multiple python versions